web/src/lib/stores/assets-store.svelte.ts

```diff
@@ -1292,12 +1324,91 @@ export class AssetStore {
     return this.buckets[0]?.getFirstAsset();
   }
 
-  async getPreviousAsset(asset: AssetResponseDto): Promise<AssetResponseDto | undefined> {
-    let bucket = await this.#getBucketInfoForAsset(asset);
+  async getPreviousAsset(
+    idable: { id: string },
+    skipTo: 'asset' | 'day' | 'month' | 'year' = 'asset',
+  ): Promise<AssetResponseDto | undefined> {
+    const bucket = this.#findBucketForAsset(idable.id);
     if (!bucket) {
       return;
     }
+    const asset = bucket.findById(idable.id);
+    if (!asset) {
+      return;
+    }
+    switch (skipTo) {
+      case 'day': {
+        return this.#getPreviousDay(asset, bucket);
+      }
+      case 'month': {
+        return this.#getPreviousMonth(asset, bucket);
+      }
+      case 'year': {
+        return this.#getPreviousYear(asset, bucket);
+      }
+      case 'asset': {
+        return this.#getPreviousAsset(asset, bucket);
+      }
+    }
+  }
+
+  async #getPreviousDay(asset: AssetResponseDto, bucket: AssetBucket) {
+    let nextDay = DateTime.fromISO(asset.localDateTime).toUTC().get('day') + 1;
+    const bucketIndex = this.buckets.indexOf(bucket);
+
+    let nextDaygroup;
+    while (nextDay <= 31) {
```

Comment:
What about months with <31 days? 👀

Reply:
I think one idea might be `.add`'ing days to the `DateTime` object until the month changes. Not sure if that's actually better readable.